### DISCUSSION TITLE
Missing: performance tips and suggestions in docs

### DISCUSSION THREAD

**Discussion by tbeason** [duckdb/duckdb#6175] [Q&A]
Perhaps the docs are focused on SQL power users, I'm not sure. I know the basics. What I don't know is what types of things are more performant. For example, suppose I have 4 columns in a table (A,B,C,D), and that columns A,B,C possibly have duplicates. Which is faster?

```
SELECT DISTINCT A, B, C FROM mytable;
```

or 

```
SELECT A, B, any_value(C) FROM mytable
GROUP BY A,B;
```

Discussions such as these, even if short, are missing from the documentation.

**Comment by Mytherin** [comment]
`DISTINCT` is equivalent to a `GROUP BY` without aggregates. So these two queries are equivalent:

```sql
-- distinct
SELECT DISTINCT A, B, C FROM mytable;
-- group by
SELECT A, B, C FROM mytable GROUP BY A, B, C;
```

If you know that certain values are keys of a table, you can use `ANY_VALUE` to optimize this, indeed. For example, if `C` does not contain duplicates this query will be faster:

```sql
SELECT A, B, any_value(C) FROM mytable
GROUP BY A,B;
```

We could indeed add more performance tuning tips to the documentation. If you have more suggestions on what to add to the documentation we would love to hear them!
